From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Sat, 5 Sep 2026 01:26:58 +0200
Subject: fix(busybox): do not install blkid unconditionally

The busybox `blkid` applet does not support the option `-o` and `-s`,
but `blkid` is used with those options in
`modules.d/70dmsquash-live-autooverlay/create-overlay.sh` and
`modules.d/70dmsquash-live/dmsquash-live-root.sh`.

So always exclude the `blkid` busybox applet.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/059d602de0e48ce8acd83f03cec443be69ba6aa9
---
 modules.d/10busybox/module-setup.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index 7f2ff85..25ff0fd 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -23,8 +23,8 @@ install() {
 
     # install busybox symlinks manually
     for _path in $($_busybox --list-full); do
-        if [[ ${_path##*/} == blkid ]] && grep -qr "blkid -o" "${udevdir}/rules.d" 2> /dev/null; then
-            # The busybox blkid applet does not support the option -o.
+        if [[ ${_path##*/} == blkid ]]; then
+            # The busybox blkid applet does not support the options -o and -s.
             # See https://github.com/vda-linux/busybox_mirror/issues/33
             continue
         fi
